its/src/test/java/com/sonar/it/shared/ScannerCliTest.java

```diff
@@ -0,0 +1,83 @@
+import com.sonar.orchestrator.container.Edition;
+import com.sonar.orchestrator.locator.MavenLocation;
+import java.io.File;
+import org.junit.Before;
+import org.junit.ClassRule;
+import org.junit.Test;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+/**
+ * Regression tests for scanning projects with the scanner-cli.
+ *
+ * Note that this uses a different orchestrator instance than {@link com.sonar.it.csharp.Tests} or {@link com.sonar.it.vbnet.Tests}
+ */
+public class ScannerCliTest {
+  private static final String RAZOR_PAGES_PROJECT = "WebApplication";
+
+  @ClassRule
+  public static final Orchestrator ORCHESTRATOR = Orchestrator.builderEnv()
+    .setSonarVersion(TestUtils.replaceLtsVersion(System.getProperty("sonar.runtimeVersion", "LATEST_RELEASE")))
+    .addPlugin(TestUtils.getPluginLocation("sonar-csharp-plugin"))
+    .addPlugin(TestUtils.getPluginLocation("sonar-vbnet-plugin"))
+    // Fixed version for the HTML plugin as we don't want to have failures in case of changes there.
+    .addPlugin(MavenLocation.of("org.sonarsource.html", "sonar-html-plugin", "3.2.0.2082"))
+    .setEdition(Edition.DEVELOPER)
+    .activateLicense()
+    .build();
+
+  @Before
+  public void init(){
+    TestUtils.deleteLocalCache();
+  }
+
+  @Test
+  public void scannerCliWithRazorPages() {
+    // As done for java, "AppSec" mode is activated to properly test sanitizers,
+    // so every unknown method is processes as a pass-through for all arguments
+    SonarScanner scanner = getSonarScanner(RAZOR_PAGES_PROJECT, "projects/" + RAZOR_PAGES_PROJECT);
+    BuildResult result = ORCHESTRATOR.executeBuild(scanner);
+
+    assertThat(result.getLogsLines(l -> l.contains("WARN")))
+      .containsExactlyInAnyOrder(
+        "WARN: No protobuf reports found. The C# files will not have highlighting and metrics.",
+        "WARN: No Roslyn issue reports were found. The C# files have not been analyzed.",
+        "WARN: Your project contains C# files which cannot be analyzed with the scanner you are using. To analyze C# or VB.NET files, you must use the Scanner for MSBuild 4.x, read more on https://redirect.sonarsource.com/doc/install-configure-scanner-msbuild.html . For any questions you may have, open a topic on https://community.sonarsource.com .",
+        "WARN: No protobuf reports found. The VB.NET files will not have highlighting and metrics.",
+        "WARN: No Roslyn issue reports were found. The VB.NET files have not been analyzed.",
+        "WARN: Your project contains VB.NET files which cannot be analyzed with the scanner you are using. To analyze C# or VB.NET files, you must use the Scanner for MSBuild 4.x, read more on https://redirect.sonarsource.com/doc/install-configure-scanner-msbuild.html . For any questions you may have, open a topic on https://community.sonarsource.com ."
+      );
+    // The HTML plugin works
+    assertThat(TestUtils.getMeasureAsInt(ORCHESTRATOR, RAZOR_PAGES_PROJECT, "violations")).isEqualTo(1);
```

Comment:
I really cannot find this manually. Can we simplify the WebApplication structure to make it obvious?